randovania/server/database.py

```diff
@@ -178,8 +178,11 @@ def layout_description(self, description: LayoutDescription | None):
     @property
     def creation_datetime(self) -> datetime.datetime:
         return datetime.datetime.fromisoformat(self.creation_date)
+    
+    def is_user_in_session(self, user: User):
+        return any(member for member in self.members if member.user == user)
```

Comment:
Not obvious, but this is terrible code!
This involves an additional query (or many? not sure) to get each member, so this is very slow.

If keeping this, best to do a direct query for the membership and handle found/not found.

Reply:
Was scrolling through documentation of peewee. Direct query would be(?!):
```
    def is_user_in_session(self, user: User):
        result = MultiplayerMembership.get_or_none((MultiplayerMembership.user == user) & (MultiplayerMembership.session == self.id))
        return result is not None
```

Reply:
Can just do `MultiplayerMembership.get_by_ids` and use a try except to catch `DoesNotExist`